sui/src/sui_commands.rs

```diff
@@ -29,53 +30,112 @@ use crate::config::{
-                working_dir,
-                config: path,
-            } => {
-                let network_path = working_dir.join("network.conf");
-                let wallet_path = working_dir.join("wallet.conf");
-                let keystore_path = working_dir.join("wallet.key");
-                let db_folder_path = working_dir.join("client_db");
-
-                if let Ok(config) = PersistedConfig::<NetworkConfig>::read(&network_path) {
-                    if !config.authorities.is_empty() {
-                        return Err(anyhow!("Cannot run genesis on a existing network, please delete network config file and try again."));
+            SuiCommand::Genesis { working_dir, force } => {
+                let sui_config_dir = &match working_dir {
+                    Some(v) => v.clone(),
+                    None => {
+                        // create default Sui config dir if not specified
+                        // on the command line and if it does not exist
+                        // yet
+                        let config_path = sui_config_dir()?;
+                        fs::create_dir_all(&config_path)?;
+                        config_path
                     }
-                }
-                let genesis_conf = if let Some(path) = path {
-                    PersistedConfig::read(path)?
-                } else {
-                    GenesisConfig::default_genesis(working_dir)?
                 };
+
+                // if Sui config dir is not empty then either clean it
+                // up (if --force/-f option was specified or report an
+                // error
+                if sui_config_dir
+                    .read_dir()
+                    .map_err(|err| {
+                        anyhow!(err).context(format!(
+                            "Cannot open Sui config dir {}",
+                            sui_config_dir.to_str().unwrap()
+                        ))
+                    })?
+                    .next()
+                    .is_some()
+                {
+                    if *force {
+                        fs::remove_dir_all(sui_config_dir).map_err(|err| {
+                            anyhow!(err).context(format!(
+                                "Cannot remove Sui config dir {}",
+                                sui_config_dir.to_str().unwrap()
+                            ))
+                        })?;
+                        fs::create_dir(sui_config_dir).map_err(|err| {
+                            anyhow!(err).context(format!(
+                                "Cannot create Sui config dir {}",
+                                sui_config_dir.to_str().unwrap()
+                            ))
+                        })?;
+                    } else {
+                        return Err(anyhow!(format!("Cannot run genesis with non-empty Sui config directory {}, please use --force/-f option to remove existing configuration", sui_config_dir.to_str().unwrap())));
```

Comment:
```suggestion
                        bail!("Cannot run genesis with non-empty Sui config directory {}, please use --force/-f option to remove existing configuration", sui_config_dir.to_str().unwrap())
```

nit: no need to use `format!` here